pkg/util/secretutil/secret.go

```diff
@@ -84,6 +102,19 @@ func getPrivateKey(data []byte) ([]byte, error) {
 		data = rest
 	}
 
+	// if both der and pemBlock are nil, then it might be a pfx cert
+	if der == nil && pemBlock == nil {
+		key, _, err := pkcs12.Decode(data, "")
```

Comment:
If we use `der` instead of new variable key, couldn't we rely on existing key parsing logic to marshal based on the key type (same as today)?

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/b2631980e2790b102dfb78afa4c90fa9c3c83fff/pkg/util/secretutil/secret.go#L87-L115

Reply:
With `pkcs12.Decode` we directly get the private key and don't have to parse it. So I am directly marshaling it and assigning it to `derKey`. Let me know if I am missing something here.

Reply:
We get the private key with `pem.Decode` too. The rest of the code here is to parse and determine if there is an error, if not we marshal it with the appropriate type (RSA/EC). By always using `MarshalPKCS1PrivateKey` here, we're assuming it's in PKCS #1 and not PKCS #8, which isn't valid. The parse code handles those scenarios and as part of the PEM block returned, it set the correct key type (`EC` or `RSA`